core/tests/vecint.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python
+
+from spt3g import core
+
+import unittest
+import os
+
+test_filename = 'inttest.g3'
+
+# Type of int needed to store the specified value.
+bit_sizes = [
+    ( 8,  0),
+    ( 8,  1),
+    ( 8, -1),
+    ( 8,  127),
+    ( 8, -128),
+    (16,  128),
+    (16, -129),
+    (16,  32767),
+    (16, -32768),
+    (32,  32768),
+    (32, -32769),
+    (32,  2147483647),
+    (32, -2147483648),
+    (64,  2147483648),
+    (64,  2147483649),
+    (64,  9223372036854775807),
+    (64, -9223372036854775808),
+]
+
+class TestVectorInt(unittest.TestCase):
+    def tearDown(self):
+        os.remove(test_filename)
+        
+    def test_serialize(self):
+        """Confirm full ranges can be saved and loaded."""
+
+        w = core.G3Writer(test_filename)
+        for isize, val in bit_sizes:
+            f = core.G3Frame()
+            f['v'] = core.G3VectorInt([val] * 10)
+            w.Process(f)
```

Comment:
It's not common to call the `Process` method directly in python, we usually just treat the pipeline module as a callable function.